force-app/main/default/classes/PlannedSalesHandler.cls

```diff
@@ -0,0 +1,69 @@
+/**
+ * @description       : 
+ * @author            : Svyatoslav Andrushchak
+ * @group             : 
+ * @last modified on  : 10-11-2024
+ * @last modified by  : Svyatoslav Andrushchak
+**/
+public class PlannedSalesHandler {
+    public static void handleAfterInsertOrUpdate(List<Planned_Sales__c> plannedSalesList) {
+        Set<Id> carTypeSet = new Set<Id>();
+        Map<Id, Planned_Sales__c> plannedSalesMap = new Map<Id, Planned_Sales__c>();
+
+        for (Planned_Sales__c ps : plannedSalesList) {
+            if (ps.Car_Type__c != null) {
+                // Car Type is a Picklist -> String values
+                carTypeSet.add(ps.Car_Type__c);
+                plannedSalesMap.put(ps.Id, ps);
+            }
+        }
+
+        // It's ok that you query line items by filtering them by Car
```

Comment:
It's ok to filter Opportunity Line Items by Car Type, but, it's not a Product2Id values,
it is Product2.Cat_Condition__c value so, from Product2Id IN :carTypeSet to Product2.Cat_Condition__c IN :carTypeSet

the other thing is that you also wanna filter by opportunity, so you select only line items u need

The easiest way to do so is no use inner child query:

List<AggregateResult> aggResults = [
            SELECT COUNT(Id) carsSold, SUM(Amount) totalAmount, CALENDAR_YEAR(CloseDate) year, CALENDAR_MONTH(CloseDate) month, **_(SELECT Id, Product2Id FROM OpportunityLineItems)_**
            FROM Opportunity
            WHERE StageName = 'Closed Won'
            AND Id IN :opportunityToCarTypeMap.keySet()
            GROUP BY CALENDAR_YEAR(CloseDate), CALENDAR_MONTH(CloseDate)
        ];